### ISSUE TITLE
Hessian2 writePlace causes StackOverflow in Dubbo 2.7.0

### ISSUE THREAD

**Issue by haiyang1985** [apache/incubator-dubbo#3766]
- [x] I have searched the [issues](https://github.com/apache/incubator-dubbo/issues) of this repository and believe that this is not a duplicate.
- [x] I have checked the [FAQ](https://github.com/apache/incubator-dubbo/blob/master/FAQ.md) of this repository and believe that this is not a duplicate.

### Environment

* Dubbo version: 2.7.0
* Operating System version: Mac OS
* Java version: 1.8

### Steps to reproduce this issue

#2031 have fixed the writePlace stackOverflow issue. The code have been merged into hessian-lite, but not in dubbo. 
Right now, the issue still there in dubbo 2.7.0. But we are not able to embed hessian-lite, as we are not able to remove hessian from dubbo(conflicts as below).
```
Found in:
    org.apache.dubbo:dubbo:jar:2.7.0:compile
    com.alibaba:hessian-lite:jar:3.2.5:compile
Duplicate classes:
    com/alibaba/com/caucho/hessian/io/HessianDebugState$DateState.class
    com/alibaba/com/caucho/hessian/io/ThrowableSerializer.class
```

1. Define a class with a writeReplace method return this
```
public class WriteReplaceReturningItself implements Serializable {

  private static final long serialVersionUID = 1L;

  private String name;

  WriteReplaceReturningItself(String name) {
    this.name = name;
  }

  public String getName() {
    return name;
  }

  /**
   * Some object may return itself for wrapReplace, e.g.
   * https://github.com/FasterXML/jackson-databind/blob/master/src/main/java/com/fasterxml/jackson/databind/JsonMappingException.java#L173
   */
  Object writeReplace() {
    //do some extra things

    return this;
  }
}
```

2. Use Hessian2Output to serialize it
```
ByteArrayOutputStream bout = new ByteArrayOutputStream();
Hessian2Output out = new Hessian2Output(bout);

out.writeObject(data);
out.flush();
```

3. Error occurs
```
java.lang.StackOverflowError
  at com.alibaba.com.caucho.hessian.io.SerializerFactory.getSerializer(SerializerFactory.java:302)
  at com.alibaba.com.caucho.hessian.io.Hessian2Output.writeObject(Hessian2Output.java:381)
  at com.alibaba.com.caucho.hessian.io.JavaSerializer.writeObject(JavaSerializer.java:226)
  at com.alibaba.com.caucho.hessian.io.Hessian2Output.writeObject(Hessian2Output.java:383)
  at com.alibaba.com.caucho.hessian.io.JavaSerializer.writeObject(JavaSerializer.java:226)
```
Pls. provide [GitHub address] to reproduce this issue.

### Expected Result
The serialization process should complete with no exception or error.

### Actual Result
```
java.lang.StackOverflowError
  at com.alibaba.com.caucho.hessian.io.SerializerFactory.getSerializer(SerializerFactory.java:302)
  at com.alibaba.com.caucho.hessian.io.Hessian2Output.writeObject(Hessian2Output.java:381)
  at com.alibaba.com.caucho.hessian.io.JavaSerializer.writeObject(JavaSerializer.java:226)
  at com.alibaba.com.caucho.hessian.io.Hessian2Output.writeObject(Hessian2Output.java:383)
  at com.alibaba.com.caucho.hessian.io.JavaSerializer.writeObject(JavaSerializer.java:226)
```

**Comment by haiyang1985** [comment]
Are we going to remove hessian from 2.7.0 or merge the PR for dubbo?

**Comment by haiyang1985** [comment]
This issue has been fixed in dubbo 2.7.1 (hessian-lite: 3.2.5). In 2.7.0, the hessian-lite is 3.2.3.
